src/applications/caregivers/helpers.js

```diff
@@ -132,3 +132,33 @@ export {
   hasSecondaryCaregiverOne,
   hasSecondaryCaregiverTwo,
 };
+
+const isSSNUnique = formData => {
+  const {
+    veteranSsnOrTin,
+    primarySsnOrTin,
+    secondaryOneSsnOrTin,
+    secondaryTwoSsnOrTin,
+  } = formData;
+
+  const allSSNs = [
+    veteranSsnOrTin,
+    primarySsnOrTin,
+    secondaryOneSsnOrTin,
+    secondaryTwoSsnOrTin,
+  ];
+
+  const allValidSSNs = allSSNs.filter(ssn => ssn !== undefined);
```

Comment:
Will this capture partially entered, invalid, SSNs?

Reply:
It technically will but the other validation function is checking for this so the edge case is covered.

Here is the other validation function:
https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/validation.js